src/main/java/TechDetectives/HobbyFinder/controllers/ListController.java

```diff
@@ -0,0 +1,64 @@
+package TechDetectives.HobbyFinder.controllers;
+import TechDetectives.HobbyFinder.Models.Data.CategoryRepository;
+import TechDetectives.HobbyFinder.Models.Data.HobbyRepository;
+import TechDetectives.HobbyFinder.Models.Category;
+import TechDetectives.HobbyFinder.Models.Hobby;
+import TechDetectives.HobbyFinder.Models.HobbyData;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import java.util.HashMap;
+import java.util.List;
+
+@Controller
+@RequestMapping(value = "list")
+public class ListController {
+
+    @Autowired
+    private HobbyRepository hobbyRepository;
+
+    @Autowired
+    private CategoryRepository categoryRepository;
+
+
+
+    static HashMap<String, String> columnChoices = new HashMap<>();
+
+    public ListController () {
+
+        columnChoices.put("all", "All");
+        columnChoices.put("category", "Category");
+
+
+    }
+
+    @RequestMapping("")
+    public String list(Model model) {
+        List categories = (List<Category>) categoryRepository.findAll();
+        model.addAttribute("categories", categories);
+
+
+
+        return "list";
+    }
+
+    @RequestMapping(value = "hobbies")
+    public String listHobbiesByColumnAndValue(Model model, @RequestParam String column, @RequestParam String value) {
```

Comment:
Is this achieved by the displaySearchResults in search controller?

Reply:
yes